### ISSUE TITLE
Use slog to provide insight on what's happening behind the scenes

### ISSUE THREAD

**Issue by rushmorem** [intecture/api#41]

**Comment by itpetey** [comment]
I think this is related to intecture/crystalball#3. Basically error reporting is not well designed or executed, which leads to very spurious errors like "file not found". The problem is that Intecture doesn't display errors in the context that they happen. All you get is the root cause. I'm not sure of the best way to remedy this intelligently. I've been looking around for better examples of error handling, especially where errors are highly nested. Suggestions welcome!

If this isn't why you wanted something like `slog`, could you explain what you are trying to achieve?

**Comment by rushmorem** [comment]
I agree that error handling and reporting needs to improve. Perhaps `error-chain` might help make this easier. However, I think `slog` will still be valuable. Unlike error reporting which only reports when an error has occurred, we can log pretty much anything important that's happening behind the scenes. This makes tracking down bugs much easier.

**Comment by itpetey** [comment]
Aah!! You made my day. `error-chain` looks awesome! I will definitely be implementing that across all projects ASAP.

I'll also have a go at implementing `slog` in the API. I'd appreciate some help/feedback in fine tuning it as I go. I'll tag commits here for your reference :)
